sparkplug-core/src/clojure/sparkplug/function.clj

```diff
@@ -73,7 +73,12 @@
                    (str/split #"/")
                    (first)
                    (symbol))
-        references (doto (HashSet.) (.add obj-ns))
+        ;; When using a piece of data as a function, such as a keyword or set,
+        ;; obj-ns will actually be a class name like `clojure.lang.Keyword`.
+        ;; Avoid marking class names as namespaces to be required
+        references (if (class? (resolve obj-ns))
+                     (HashSet.)
+                     (doto (HashSet.) (.add obj-ns)))
```

Comment:
It may be clearer to move the side-effect outside of the let binding now:
```clojure
(let [,,,
      references (HashSet.)
      visited (HashSet.)]
  ;; ...exposition...
  (when-not (class? (resolve obj-ns))
    (.add references obj-ns))
  ,,,)
```